openedx/core/djangoapps/discussions/serializers.py

```diff
@@ -282,23 +294,140 @@ def _update_plugin_configuration(
         """
         Create/update legacy provider settings
         """
+        plugin_configuration = validated_data.pop('plugin_configuration', {})
         updated_provider_type = validated_data.get('provider_type') or instance.provider_type
         will_support_legacy = bool(
             updated_provider_type == 'legacy'
         )
         if will_support_legacy:
-            course_key = instance.context_key
-            course = get_course_by_id(course_key)
             legacy_settings = LegacySettingsSerializer(
-                course,
+                self._get_course(),
                 context={
                     'user_id': self.context['user_id'],
                 },
-                data=validated_data.get('plugin_configuration', {}),
+                data=plugin_configuration,
             )
             if legacy_settings.is_valid(raise_exception=True):
                 legacy_settings.save()
-            instance.plugin_configuration = {}
+            instance.plugin_configuration = {
+                "group_at_subsection": plugin_configuration.get("group_at_subsection", False)
+            }
         else:
-            instance.plugin_configuration = validated_data.get('plugin_configuration') or {}
+            instance.plugin_configuration = plugin_configuration
+        return instance
+
+    def _update_course_configuration(
+        self,
+        instance: DiscussionsConfiguration,
+        validated_data: dict,
+    ) -> DiscussionsConfiguration:
+        """
+        Update configuration settings that are stored in the course.
+        """
+        save = False
+        updated_provider_type = validated_data.get('provider_type') or instance.provider_type
+        for key in self.Meta.course_fields:
+            value = validated_data.get(key)
+            # Delay loading course till we know something has actually been updated
+            if value is not None and value != getattr(instance, key):
+                self._get_course().discussions_settings[key] = value
```

Comment:
I know it doesn't load the course every time, but let's make it simpler and store that value in a variable `course = self._get_course()` to make it more readable.

Reply:
We can do that but in that case every call to this API will load the course, even if it doesn't need to.

Reply:
Ah sure! I thought you meant loading it once up top and using that. Will update.